consensus/parlia/parlia.go

```diff
@@ -1137,6 +1143,30 @@ func (p *Parlia) Finalize(chain consensus.ChainHeaderReader, header *types.Heade
 			return err
 		}
 	}
+
+	if p.chainConfig.IsOnFusion(header.Number) {
+		err := p.initFusionContract(state, header, cx, txs, receipts, systemTxs, usedGas, false)
+		if err != nil {
+			log.Error("init fusion contract failed")
+		}
+	}
+
+	// update validators every day
+	if p.chainConfig.IsFusion(header.Number) {
+		parent := chain.GetHeaderByHash(header.ParentHash)
+		if parent == nil {
+			return errors.New("parent not found")
+		}
+		if time.Unix(int64(parent.Time), 0).Day() == time.Unix(int64(header.Time), 0).Day()-1 {
```

Comment:
The code is not friend to test, how will you deal with that?

Reply:
I will set it to `time.Unix(int64(header.Time), 0).Minute() % 10 == 0 ` or something like that.